packages/expo-router/src/fork/extractPathFromURL.ts

```diff
@@ -43,26 +42,36 @@ function extractExactPathFromURL(url: string): string {
 }
 
 /** Major hack to support the makeshift expo-development-client system. */
-function isExpoDevelopmentClient(url: URL<Record<string, string | undefined>>): boolean {
+function isExpoDevelopmentClient(url: URL): boolean {
   return !!url.hostname.match(/^expo-development-client$/);
 }
 
 function fromDeepLink(url: string): string {
-  // This is for all standard deep links, e.g. `foobar://` where everything
-  // after the `://` is the path.
-  const res = new URL(url, true);
+  let res: URL;
+  try {
+    // This is for all standard deep links, e.g. `foobar://` where everything
+    // after the `://` is the path.
+    res = new URL(url);
+  } catch {
+    return url;
+  }
 
   if (isExpoDevelopmentClient(res)) {
-    if (!res.query || !res.query.url) {
+    if (!res.search || !res.searchParams.get('url')) {
```

Comment:
searchParams is always defined, can just do:
```suggestion
    if (!res.searchParams.get('url')) {
```